code-challenges.test.js

```diff
@@ -21,11 +22,62 @@
 // Example input: 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("fibSequence", () => {
 
-// b) Create the function that makes the test pass.
+    // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+    it("takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence", () => {
 
+        //an expect method, nested within the test block, calling on the fibSequence() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+        expect(fibSequence(6)).toEqual([1, 1, 2, 3, 5, 8])
+        expect(fibSequence(10)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+    })
+})
+// -- Got the red --  
 
-// --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+// b) Create the function that makes the test pass.
+// create a function that takes in an number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence
+function fibSequence(number) {
+// declare the arrays
+    var fibArray = [1, 1];
+// declare another array, find out we don't need this array, will delete this array when we refactoring
+    var newFibArray
+    //build a if statement check the number is greater than 2 or not
+    if (number > 2) {
+        // if greater than 2 is true, we gonna make a for loop, let i starts with 2, because the first 2 values in the array are always [1,1]
+        // and we only care about the number greater than 2
+        for (let i = 3 - 1; i < number; i++) {
```

Comment:
I get where you are going with declaring i = 3-1, but I think you could just call it i = 2.